runtime/tests/org.eclipse.core.tests.runtime/src/org/eclipse/core/tests/runtime/jobs/JobGroupTest.java

```diff
@@ -515,31 +516,30 @@ public void testJoinWithTimeout() {
 		final long duration[] = {-1};
 
 		Thread t = new Thread(() -> {
-			status.set(0, TestBarrier2.STATUS_START);
 			try {
 				long start = now();
 				firstJobGroup.join(timeout, null);
 				duration[0] = now() - start;
-			} catch (OperationCanceledException | InterruptedException e) {
-				// ignore
+			} catch (InterruptedException e1) {
+				fail("Unexpected interrupt occurred while joining job group");
+			} catch (OperationCanceledException e2) {
+				fail("Join was unexpectedly canceled");
 			}
-			status.set(0, TestBarrier2.STATUS_DONE);
+			barrier.setStatus(TestBarrier2.STATUS_DONE);
 		});
-
 		// Start the thread that will join the first job group and be blocked until the join call is returned.
 		t.start();
-		TestBarrier2.waitForStatus(status, 0, TestBarrier2.STATUS_START);
-		int i = 0;
-		for (; i < 11; i++) {
-			if (status.get(0) == TestBarrier2.STATUS_DONE) {
-				// Verify that the join call is blocked for at least for the duration of given timeout.
-				assertTrue("1.0 duration: " + Arrays.toString(duration) + " timeout: " + timeout, duration[0] >= timeout);
-				break;
-			}
-			sleep(100);
-		}
-		// Verify that the join call is returned is finished with in reasonable time of 1100 ms (given timeout + 100ms).
-		assertTrue("2.0", i < 11);
+
+		barrier.waitForStatus(TestBarrier2.STATUS_DONE);
+		// Verify that thread is done
+		assertNotEquals("Waiting for join to time out failed", -1, duration[0]);
+		assertEquals("Group with long running jobs should still be active after join timed out", JobGroup.ACTIVE,
+				firstJobGroup.getState());
+		// Verify that join call is blocked for at least the duration of given timeout
+		String durationAndTimoutMessage = "duration: " + duration[0] + " timeout: " + timeout;
+		assertTrue("Join did not run into timeout; " + durationAndTimoutMessage, duration[0] >= timeout);
+		assertTrue("Join took significantly longer than timeout; " + durationAndTimoutMessage,
```

Comment:
when the build hangs for a random second due to other things this can randomly fail. better avoid this check

Reply:
Yes, thanks! In case of a regression regarding timeout functionality, the test will fail anyway as the barrier waiting for a status change times out (but then in a reasonable amount of time rather than the 400ms stated here).

I have removed the check. We should then remove the analogous check in `JobTest` as well, as it may randomly fail for the same reasons. I'll provide a follow-up PR for that.

Reply:
@jukzi are you okay with the removal of the check?